### ISSUE TITLE
module.hot.invalidate is not a function

### ISSUE THREAD

**Issue by shaodahong** [pmmmwh/react-refresh-webpack-plugin#157]
If i change `Class` -> `Function` or `Function` -> `Class`, error overlay by

![image](https://user-images.githubusercontent.com/14831261/88364400-5c53ec80-cdb5-11ea-9c52-4ca657e21803.png)

```jsx
import React from 'react';

// export default class Text extends React.Component {
//   render() {
//     return <div>{this.props.children}</div>
//   }
// }

export default function Text({ children }) {
  return <div>{children}</div>;
}

```

**Comment by shaodahong** [comment]
Base on cra example

**Comment by bebraw** [comment]
Are you using webpack 4.43.0 or newer? The method is introduced there and it doesn't exist before.

**Comment by shaodahong** [comment]
In my local webpack version `4.42.0`, and i upgrade and try

maybe we need upgrade this demo webpack version?

https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/examples/create-react-app/package.json

**Comment by bebraw** [comment]
Yeah, I noticed at least one of the demos is broken. Likely it's not a bad idea to update webpack version in each. I would also add some kind of snapshot test against their output to avoid breakage (simple to do).

**Comment by shaodahong** [comment]
Thanks

![Kapture 2020-07-24 at 16 12 05](https://user-images.githubusercontent.com/14831261/88372458-85ca4380-cdc8-11ea-9ab9-eb32263d3485.gif)

**Comment by pmmmwh** [comment]
Thanks for spotting this. This happens because for 0.4.0+ versions we will drop support for Webpack versions below 4.43.0 - in theory the plugin could be made backward compatible, falling back to `window.location.reload`, but that heuristic causes a lot of problem when the users are not using the browser as a renderer, so I am hesitant to do it.

**Comment by shaodahong** [comment]
> Thanks for spotting this. This happens because for 0.4.0+ versions we will drop support for Webpack versions below 4.43.0 - in theory the plugin could be made backward compatible, falling back to `window.location.reload`, but that heuristic causes a lot of problem when the users are not using the browser as a renderer, so I am hesitant to do it.

can be check support `window.location.reload`?

**Comment by pmmmwh** [comment]
> can be check support `window.location.reload`?

The main thing is, it would still not work for other non-browser renderers, and it provides quite a different experience for non 4.43+ users (a lot more cases would force refresh).

Maybe we can somehow pull all the fixes in 0.4 into 0.3 ...
